kernel/aster-nix/src/thread/thread_table.rs

```diff
@@ -4,18 +4,18 @@ use super::{Thread, Tid};
 use crate::prelude::*;
 
 lazy_static! {
-    static ref THREAD_TABLE: Mutex<BTreeMap<Tid, Arc<Thread>>> = Mutex::new(BTreeMap::new());
+    static ref THREAD_TABLE: SpinLock<BTreeMap<Tid, Arc<Thread>>> = SpinLock::new(BTreeMap::new());
 }
```

Comment:
I use `SpinLock` here because it's necessary to access the THREAD_TABLE when all processors initiate the first task, and at this point, without a current task, sleeping is not an option. Although accessing THREAD_TABLE with a `Mutex` later on won't pose a problem, I haven't yet come up with a good way to standardize this behavior.